intruder_alarm/src/stack/mod.rs

```diff
@@ -75,125 +76,123 @@ pub trait Linked: Sized {
 // Implementations
 //-----------------------------------------------------------------------------
 
-// ===== impl List =====
+// ===== impl Stack =====
 
-impl<T, Node, R> List<T, Node, R> {
-    /// Create a new `List` with 0 elements.
+impl<T, Node, R> Stack<T, Node, R> {
+    /// Create a new `Stack` with 0 elements.
     pub const fn new() -> Self {
-        List {
-            head: Link::none(),
-            len: 0,
+        Stack {
+            top: Link::none(),
+            size: 0,
             _elem_ty: PhantomData,
             _ref_ty: PhantomData,
         }
     }
 
-    /// Returns the length of the list.
+    /// Returns the size of the stack.
     #[inline]
-    pub fn len(&self) -> usize {
-        self.len
+    pub fn size(&self) -> usize {
```

Comment:
Let's keep this `len()`, that seems to be the convention for collection types in Rust.

Reply:
Let me push this change. Good to know about Rust's conventions (although I should have checked those myself).